app/models/user.rb

```diff
@@ -1,4 +1,18 @@
 class User < ApplicationRecord
   attr_accessor :display_name, :admin, :roles
   validates :login, presence: true
+
+  def self.from_jwt(data = { })
+    id, login = data&.values_at 'id', 'login'
```

Comment:
The token's being generated by Panoptes, so probably not. This method also handles the anonymous user case. I suppose it could if the JWT is edited manually by either the user or the front end viewer/editor, but it'd be a bit of work to edit that'd just end up with a valid request that has no permissions.

That said, it does seem like it should be the controller conditionally setting `@current_user` based on whether the token exists, and not the responsibility of `User#from_jwt` to decide. So I moved the token.present? question to `ApplicationController#set_user` and refactored that JWT method to throw an error if it gets called with bad data.